scipy/stats/tests/test_distributions.py

```diff
@@ -896,6 +897,44 @@ def test_mean_small_p(self):
         assert_allclose(m, 1.000000005)
 
 
+class TestGumbel_r(object):
+    def setup_method(self):
+        np.random.seed(1234)
+
+    @pytest.mark.parametrize("loc_rvs,scale_rvs", [np.random.rand(2)])
+    def test_fit(self, loc_rvs, scale_rvs):
+        data = stats.gumbel_r.rvs(size=100, loc=loc_rvs, scale=scale_rvs)
+
+        # test that result of fit method is the same as optimization
+        def func(vals, data):
+            a, b = vals
+            ndata = len(data)
+            x1 = data.mean() - ((np.sum(data * np.exp(- data / b))) /
+                                np.sum(np.exp(- data / b))) - b
+            x2 = - b * np.log(np.sum(np.exp(-data / b)) / ndata) - a
+            return x1, x2
+
+        expected_solution = root(func, stats.gumbel_r._fitstart(data), args=(
+            data,)).x
+        fit_method = stats.gumbel_r.fit(data)
+
+        # other than computational variances, the fit method and the solution
+        # to this system of equations are equal
+        assert_allclose(fit_method, expected_solution, atol=1e-30)
+
+    @pytest.mark.parametrize("loc_rvs,scale_rvs", [np.random.rand(2)])
+    def test_fit_comp_optimizer(self, loc_rvs, scale_rvs):
+        data = stats.gumbel_r.rvs(size=100, loc=loc_rvs, scale=scale_rvs)
+
+        # obtain objective function to compare results of the fit methods
+        args = [data, (stats.gumbel_r._fitstart(data),)]
+        func = stats.gumbel_r._reduce_func(args, {})[1]
+
+        # test that gumbel_r fit method is better than super method
+        _assert_lessthan_loglike(stats.gumbel_r, data, func)
+
+
```

Comment:
I think it's fine that we're not testing for e.g. invalid values of fixed parameters because all of that is checked by the superclass `fit` method. I think the only test that should be required specific to this override of `fit` is the one you have above. ✓